vspace/src/main/webapp/WEB-INF/views/staff/search/staffSearch.html

```diff
@@ -0,0 +1,135 @@
+<head>
+<script th:src="@{/resources/bootpag/js/bootpag.min.js}"></script>
+<script th:inline="javascript">
+	$(document).ready(function($) {
+		var modelAttributeValue = [[${searchResults}]];
+        if(modelAttributeValue.length != 0) {
+            $("#noResult").hide();
+            $("#resultDisplay").show();
+        }
+        else {
+            $("#noResult").show();
+            $("#resultDisplay").hide();
+        }
+		$(function() {
+			
+			$('[data-toggle="tooltip"]').tooltip()
+		});
+		
+		 $('#page-selection').bootpag({
+		        total: [(${totalPages})],
+		        page: [(${currentPageNumber})],
+		        maxVisible: 10,
+		        leaps: true,
+		        firstLastUse: true,
+		        first: '|<',
+		        last: '>|',
+		        wrapClass: 'pagination',
+		        activeClass: 'active',
+		        disabledClass: 'disabled',
+		        nextClass: 'next',
+		        prevClass: 'prev',
+		        lastClass: 'last',
+		        firstClass: 'first'
+		    }).on("page", function(event, num){
+		        window.location.assign("./"+num+"?searchText="+[[${searchWord}]]);
+		    });
+
+	});
+</script>
+<style>
+.pagination>li>a {
+    border: 1px solid var(--primary);
+    padding: 5px;
+    color: var(--primary);
+    font-size: 15px;
+}
+
+.pagination>li.active>a {
+    background: var(--primary);
+    color: #fff;
+    font-size: 15px;
+}
+</style>
+</head>
+<body>
+    <div class="main-content" layout:fragment="content">
+
+        <div class="">
+            <div><h3>Search Results</h3>
+            <h5>Keyword: <grey><b>[[${searchWord}]]</b></grey></h5></div>
+            <div id="noResult">
+            <h5><center>No result to display.</center></h5>
+            </div>
+            <div id="resultDisplay">
+                <div th:each="contents: ${searchResults}">
+                    <div th:id="${contents.id}"
+                        th:if="${#strings.equals(contents['class'].simpleName,'Space')}">
+                        <a th:href="@{'/staff/space/'+${contents.id}}"
+                            class="textDivStaff tab  btn btn-light"
+                            style="width: 100%"> <span
+                            class="card-title"
+                            style="white-space: pre-wrap; line-break: anywhere;"><img
+                            style="border-radius: 25px; float: left; width: 150px; height: 150px;"
+                            th:src="@{'/api/image/'+${contents?.image?.id}}" alt="Space image" /><span 
+                            style="color: var(--dark-grey)">  Space:</span> [[${contents.name}]]<br><span 
+                            style="color: var(--dark-grey)">  Description:</span> [[${contents.description}]]</span>
+                        </a>
+                    </div>
+
+                    <div th:id="${contents.id}"
+                        th:if="${#strings.equals(contents['class'].simpleName,'Module')}">
+                        <th:block th:text="${conIndex.setLength(0)}" />
```

Comment:
what is conIndex for?

Reply:
Use of conindex is mentioned in the above the above comment's reply section.

Reply:
Instead of this very hard to understand solution,  just add methods to the slide class that return the first image or first text block.

Reply:
this is not resolved. conIndex is still here.